### DISCUSSION TITLE
How to use a different linker?

### DISCUSSION THREAD

**Discussion by stefanos82** [transmission/transmission#4608] [Q&A] [answered]
Since GCC 12 (I believe since 12.1 to be more precise), we can use `-fuse-ld=` flag to choose a different linker.

[mold](https://github.com/rui314/mold) is an extremely fast linker that I have tested on some other projects and compiles really fast comparing the standard `ld` linker that GCC ships; `gold` is a bit better than `ld`, but it cannot get compared with `mold`.

Currently I use `cmake -G Ninja -DCMAKE_BUILD_TYPE=RelWithDebInfo ..`, but I would like it to use `mold` as well.

Any suggestions?

**Comment by mikedld** [comment, marked as answer]
That's more of a CMake-specific question. I assume you could try adding something like `-DCMAKE_EXE_LINKER_FLAGS=-fuse-ld=mold` ([ref](https://cmake.org/cmake/help/latest/variable/CMAKE_EXE_LINKER_FLAGS.html)) to your configuration command, provided that compiler itself is used to link, or adjust `CMAKE_CXX_LINK_EXECUTABLE` ([ref](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_LINK_EXECUTABLE.html)) variable if not.

**Reply by stefanos82** [reply]
Interesting! Let me try the first one and will come back with further info.

**Reply by stefanos82** [reply]
Seems like it works, but it's so darn slow to finish compilation!

Why does it take a century, so to speak, to finish `ninja transmission-gtk`?

I would not dare to run plain `ninja` as it would take hours to finish on my system.

**Reply by mikedld** [reply]
I'd say that's a whole different question. Could be that you have a suitable version of clang-tidy installed, and it's being picked up and used (which is what's happening by default), considerably slowing the compilation down. If that's the case, passing `-DRUN_CLANG_TIDY=OFF` will help.

**Reply by stefanos82** [reply]
Oh blimey, I remember this from the past that I had in my old notes!

Indeed it was the stupid `clang-tidy` flag...it finished in 2-3 minutes.

What an awkward behavior :laughing: , thank you @mikedld !
